pkg/interfaces/contracts/vault/VaultTypes.sol

```diff
@@ -105,6 +105,56 @@ struct PoolRoleAccounts {
 }
 
+/*******************************************************************************
+                                   Tokens
+*******************************************************************************/
+
+// Note that the following tokens are unsupported by the Vault. This list is not meant to be exhaustive, but covers
+// many common types of tokens that will not work with the Vault architecture. (See https://github.com/d-xo/weird-erc20
+// for examples of token features that are problematic for many protocols.)
+//
+// * Rebasing tokens (e.g., aDAI). The Vault keeps track of token balances in its internal accounting; any token whose
+//   balance changes asynchronously (i.e., outside a swap or liquidity operation), would get out-of-sync with this
+//   internal accounting. This category would also include "airdrop" tokens, whose balances can change unexpectedly.
+//
+// * Double entrypoint (e.g., old Synthetix tokens, now fixed). These could likewise bypass internal accounting by
+//   registering the token under one address, then accessing it through another. This is especially troublesome
+//   in v3, with the introduction of ERC4626 buffers.
+//
+// * Fee on transfer (e.g., PAXG). The Vault issues credits and debits according to given and calculated token amounts,
+//   and settlement assumes that the send/receive transfer functions transfer exactly the given number of tokens.
+//   If this is not the case, transactions will not settle. Unlike with the other types, which are fundamentally
+//   incompatible, it would be possible to design the Vault to handle this - but it was thought this was too much
+//   complexity and gas to support an edge case.
+//
+// * Tokens with greater than 18 decimals (e.g., YAM-V2). The Vault handles token scaling: i.e., handling I/O for
+//   amounts in native token decimals, but doing calculations with full 18-decimal precision. This requires reading
+//   and storing the decimals for each token. Since virtually all tokens are 18 or fewer decimals, and we have limited
+//   storage space, 18 was a reasonable maximum. Unlike the other types, this is enforceable by the Vault. Attempting
+//   to register such tokens will revert with `InvalidTokenDecimals`. Of course, we must also be able to read the token
+//   decimals, so the Vault only supports tokens that implement `IERC20Metadata.decimals`, and return a value less than
+//   or equal to 18.
+//
+// These types of tokens are supported but discouraged, as they don't tend to play well with AMMs generally.
+//
+// * Very low-decimal tokens (e.g., GUSD). The Vault has been extensively tested with 6-decimal tokens (e.g., USDC),
+//   but going much below that may lead to unanticipated effects due to precision loss, especially with smaller trade
+//   values. Also, the Vault enforces a minimum trade size for security. Using a minimum of 1e6 corresponds to 10,000
+//   GUSD, making it impossible to trade small amounts of low-decimal tokens.
```

Comment:
The 1e6 limit corresponds to the scaled amount. If GUSD has 2 decimals, 1 wei is 1e16 when scaled by the decimal scaling factor. It shouldn't ever hit the minimum trade amount.